plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/Realtime.tsx

```diff
@@ -0,0 +1,150 @@
+import { Icon, Switch } from "@web/sharedComponents";
+import { styled } from "@web/theme";
+import { useCallback, useEffect, useRef, useState } from "react";
+
+import { BaseFieldProps } from "../types";
+
+const Realtime: React.FC<BaseFieldProps<"realtime">> = ({ value, editMode, onUpdate }) => {
+  const [intervalInSecond, setIntervalValue] = useState<number>(value.updateInterval ?? 30);
+  const [timer, setTimer] = useState<string>("00:00:00");
+  const [enableUpdate, changeUpdateState] = useState<boolean>(true);
+  const Ref = useRef<any>(null);
+
+  const getTimeRemaining = useCallback((interval: number) => {
+    const seconds = Math.floor(interval % 60);
+    const minutes = Math.floor((interval / 60) % 60);
+    const hours = Math.floor((interval / 60 / 60) % 24);
+    if (interval > 0) {
+      setTimer(
+        (hours > 9 ? hours : "0" + hours) +
+          ":" +
+          (minutes > 9 ? minutes : "0" + minutes) +
+          ":" +
+          (seconds > 9 ? seconds : "0" + seconds),
+      );
+    }
+  }, []);
+
+  const startTimer = useCallback(
+    (interval: number) => {
+      if (Ref.current) clearInterval(Ref.current);
+      const id = setInterval(() => {
+        if (enableUpdate) {
+          getTimeRemaining(interval);
+          interval--;
+          if (interval === 0) interval = intervalInSecond;
+        }
+      }, 1000);
+
+      Ref.current = id;
+    },
+    [enableUpdate, getTimeRemaining, intervalInSecond],
+  );
+  const clearTimer = useCallback(() => {
+    if (Ref.current) clearInterval(Ref.current);
+  }, []);
+
+  const handleChangeupdateState = useCallback(() => {
+    changeUpdateState(!enableUpdate);
+    setTimer("00:00:30");
+    if (enableUpdate) startTimer(intervalInSecond);
```

Comment:
Since you are doing !enableUpdate above, I think this will start the timer when enableUpdate is false because if it is **becoming** true, in this if statement it will still be false until the next time it is called.

Reply:
but starting the timer is set only when enableUpdate is true

Reply:
c9e5706fc4bb3535ff518c320d648fb4f2ea4b1b

Reply:
That is exactly what I mean. In this handleChangeUpdateState function, the first thing you do is `changeUpdateState(!enableUpdate);`. This, if enableUpdate was false, will make it true, right? But it won't be true in this useCallback until the next time the useCallback is used. Which means the enableUpdate in ->`if(enableUpdate) startTimer(intervalInSecond)` is still false. 
Understand?

Reply:
you mean I should change the state `changeUpdateState(!enableUpdate); `after checking the `if(enableUpdate)startTimer(intervalInSecond)` but even if I do that there will be a ms while checking the current state , there is another solution to use Ref for example , is that what you mean ?